api/admin/controller.py

```diff
@@ -462,6 +555,74 @@ def edit(self, identifier_type, identifier):
             staff_edition.series_position = new_series_position
             changed = True
 
+        new_medium = flask.request.form.get("medium")
+        if new_medium:
+            if new_medium not in Edition.medium_to_additional_type.keys():
+                self._db.rollback()
+                return UNKNOWN_MEDIUM.detailed(
+                    _("Medium %(medium)s is not one of the known media.",
+                      medium=new_medium))
+            staff_edition.medium = new_medium
+            changed = True
+
+        new_language = flask.request.form.get("language")
+        if new_language:
+            new_language = LanguageCodes.string_to_alpha_3(new_language)
+            if not new_language:
+                self._db.rollback()
+                return UNKNOWN_LANGUAGE
+        if new_language != staff_edition.language:
+            staff_edition.language = new_language
+            changed = True
+
+        new_publisher = flask.request.form.get("publisher")
+        if new_publisher != staff_edition.publisher:
+            if staff_edition.publisher and not new_publisher:
+                new_publisher = NO_VALUE
+            staff_edition.publisher = unicode(new_publisher)
+            changed = True
+
+        new_imprint = flask.request.form.get("imprint")
+        if new_imprint != staff_edition.imprint:
+            if staff_edition.imprint and not new_imprint:
+                new_imprint = NO_VALUE
+            staff_edition.imprint = unicode(new_imprint)
+            changed = True
+
+        new_issued = flask.request.form.get("issued")
+        if new_issued:
+            try:
+                new_issued = datetime.strptime(new_issued, '%Y-%m-%d')
+            except ValueError:
+                self._db.rollback()
+                return INVALID_DATE_FORMAT
+        if new_issued != staff_edition.issued:
+            staff_edition.issued = new_issued
+            changed = True
+
+        # TODO: This lets library staff add a 1-5 rating, which is used in the
+        # quality calculation. However, this doesn't work well if there are any
+        # other measurements that contribute to the quality. The form will show
+        # the calculated quality rather than the staff rating, which will be
+        # confusing. It might also be useful to make it more clear how this
+        # relates to the quality threshold in the library settings.
```

Comment:
We could change the quality calculation so that if a rating is present from a certain source, no other ratings are considered.

Reply:
Do you think we should do that? How many other sources do we get ratings from?

We could also do something like we do with classifications where we show all the sources and weights.

Reply:
The main other rating source is the distributor itself, e.g. Overdrive.

I like the idea of showing all the sources and weights, but weighting library staff opinion very heavily.

Reply:
I agree but I'm not sure how to present this information, so for now I think I'd like to expose the staff rating in the admin OPDS annotator and use that to populate the form rather than quality, and also give the staff rating a high weight.

Reply:
Sounds good.